### PR TITLE
CheckBox inline bug

### PR DESCRIPTION
**Pull request by JohnStuartBarrie**
# Description
Checkbox should not cover the last two letters and the full stop of the Help field text (see screenshots).

### CODE REVIEW

File: src/components/checkbox/checkbox.scss [hunk 1]
```diff
@@ -98,7 +98,7 @@
     display: inline;
     left: 21px;
     margin-left: -21px;
```
Comment (andrewjtait) [inline src/components/checkbox/checkbox.scss]: It looks like the `left` and `margin-left` properties may be the issue here, doesn't make much sense why they would be overriding each other - and the `left` prop is probably what is throwing the input off.

Would it work if you left `margin-right` as `5px`, changed `margin-left` to `0`, and deleted `left`?

Comment (JohnStuartBarrie) [reply src/components/checkbox/checkbox.scss]: I tried this actually and discover the left seemed important for something. Can't remember what. I will have another look.

File: src/components/checkbox/checkbox.scss [hunk 2]
```diff
@@ -96,10 +96,9 @@
 
   &.carbon-checkbox__help-text--inline {
     display: inline;
-    left: 21px;
-    margin-left: -21px;
-    margin-right: 5px;
+    left: 0;
```
Comment (andrewjtait) [inline src/components/checkbox/checkbox.scss]: screen shot 2018-01-19 at 15 33 28

setting `left: 0` and removing `margin-left` breaks the non-reversed version. Looks like it works if you remove `left` and set `margin-left` to `0`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+# 3.0.0
+
+## Bug Fixes
+
+* Checkbox no longer overlays the end of the Help field text when the Reverse box is ticked
```
Comment (ianoxley) [inline CHANGELOG.md]: Minor - replace 'when the Reverse box is ticked' with 'when the `reverse` prop is set to `true`.

File: src/components/checkbox/checkbox.scss (hunk 2 above)
Comment (JohnStuartBarrie) [reply src/components/checkbox/checkbox.scss]: Maybe I am blind (but I can't see what broke).I removed both. We should need either. right?

Comment (andrewjtait) [reply src/components/checkbox/checkbox.scss]: The gap between the icon and the help text is too wide

Comment (andrewjtait) [reply src/components/checkbox/checkbox.scss]: `margin-left: 0` ensures it is correct

**Pull request merged**